lib/App/Pinto/Command/add.pm

```diff
@@ -232,6 +233,51 @@ to work on those distributions, in the event that you need to patch them
 in the future.  Be aware that most distributions do not actually declare
 their development prerequisites.
 
+=item --verify-upstream=[0-5]
+
+=item -Z [0-5]
+
+=item -Z ... -ZZZZZ
+
+!! THIS OPTION IS EXPERIMENTAL !!
+
+Require upstream distribution files to be verified before operating on them.
+Repeated use of this argument (up to 5) requires the upstream verification to
+be progressively more strict.  You can also set the verification level
+explicitly, e.g.,
+
+    --verify-upstream=3
+
+At level 0, no verification is performed. This may be useful if you need to
+override the verification level set earlier, say, in a script.
+
+At level 1, we verify the distributions checksum using the upstream CHECKSUMS
+file.  This is the default level.
+
+At level 2, we also verify the signature on the upstream CHECKSUMS file if it
+has one.  Warnings about unknown or untrusted PGP keys relating to that file
+are printed.
+
+At level 3, we also require upstream CHECKSUMS files to be signed.  Warnings
+about unknown or untrusted PGP keys relating to that file are now considered
+fatal.
+
+At level 4, we also verify the unpacked distribution using the embedded
+SIGNATURE file if it exists.  Warnings about unknown or untrusted PGP keys
+relating to that file are printed.
+
+At level 5, warnings about unknown or untrusted PGP keys relating to embedded
+SIGNATURE files are now considered fatal.
+
+Note that none of these checks are applied to LOCAL distributions, i.e.,
+distributions that do not have an upstream CHECKSUMS file.
+
+The impact of this option will largely depend on the your chosen upstream
+repositories and state of your current keyring.  Consider using a dedicated
+keyring/trustdb via the C<PINTO_GNUPGHOME> environment variable.  See the
+documentation for the L<verify|App::Pinto::Command::verify> command for the
+rationale and an example.
+
```

Comment:
Either level 2 or 3 depending on how you have your keyring setup.  I'm
currently using level 3 because I've gone to the trouble of verifying the
PAUSE Batch signing key. Levels 4 and 5 depend on how you regard embedded
SIGNATURE files -- its a bit contentious.  The --verify flag for cpanm does
a mixture of 2 and 4.


Kahlil (Kal) Hodgson                       GPG: C9A02289
Head of Technology                         (m) +61 (0) 4 2573 0382
DealMax Pty Ltd

Suite 1416
401 Docklands Drive
Docklands VIC 3008 Australia

"All parts should go together without forcing.  You must remember that
the parts you are reassembling were disassembled by you.  Therefore,
if you can't get them together again, there must be a reason.  By all
means, do not use a hammer."  -- IBM maintenance manual, 1925

On 6 February 2015 at 16:10, Jeffrey Ryan Thalhammer <
notifications@github.com> wrote:

> In lib/App/Pinto/Command/add.pm
> <https://github.com/thaljef/Pinto/pull/182#discussion_r24222971>:
>
> > +At level 4, we also verify the unpacked distribution using the embedded
> > +SIGNATURE file if it exists.  Warnings about unknown or untrusted PGP keys
> > +relating to that file are printed.
> > +
> > +At level 5, warnings about unknown or untrusted PGP keys relating to embedded
> > +SIGNATURE files are now considered fatal.
> > +
> > +Note that none of these checks are applied to LOCAL distributions, i.e.,
> > +distributions that do not have an upstream CHECKSUMS file.
> > +
> > +The impact of this option will largely depend on the your chosen upstream
> > +repositories and state of your current keyring.  Consider using a dedicated
> > +keyring/trustdb via the C<PINTO_GNUPGHOME> environment variable.  See the
> > +documentation for the L<verify|App::Pinto::Command::verify> command for the
> > +rationale and an example.
> > +
>
> Which one of these levels would be appropriate if the upstream host was a
> CPAN mirror.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/thaljef/Pinto/pull/182/files#r24222971>.
>